gnomad/resources/resource_utils.py

```diff
@@ -38,13 +39,14 @@ def __init__(
 
         if (
             path is not None
-            and expected_file_extensions
-            and not any(path.endswith(ext) for ext in expected_file_extensions)
+            and self.expected_file_extensions
+            and not any(path.endswith(ext) for ext in self.expected_file_extensions)
         ):
             logger.warning(
-                "Created the following {} with a path that doesn't end with {}: {}".format(
-                    self.__class__.__name__, " or ".join(expected_file_extensions), self
-                )
+                "Created the following %s with a path that doesn't end with %s: %s",
+                self.__class__.__name__,
+                " or ".join(self.expected_file_extensions),
+                self,
```

Comment:
This resolves a [logging-not-lazy](https://pylint.readthedocs.io/en/latest/technical_reference/features.html#logging-checker-messages) message from Pylint.

Reply:
I'm not sure I follow why there's an error here from Pylint -- it looks like there's f-string formatting on line 55. Is this just due to the use of string variables in the logger? It would be nice to be have a consistent style across the repo. Does Pylint still object if we do `"Created the following {self.__class__.__name__} with a path that doesn't end with {" or ".join(expected_file_extensions)}: {self}"`?

Reply:
The warning is specifically about f-strings in logger calls. Logging with `logger.info("message about %s", obj)` instead of `logger.info(f"message about {obj}")` or `logger.info("message about %s" % obj)` avoids a crash if there's an error formatting the message and only constructs the message if it will actually be output.

https://medium.com/swlh/why-it-matters-how-you-log-in-python-1a1085851205

Reply:
huh, I didn't know this. If we are switching to this method we should make sure the rest of the team is aware because I think most of us use f-strings in logger calls.

Reply:
Ideally, we could fix all the Pylint warnings and then start enforcing them on PR checks.

https://github.com/broadinstitute/gnomad_methods/blob/e0f433aee33adefce837e8406f0eeb6323a71796/.github/workflows/ci.yml#L30-L31

Reply:
Thanks, Nick -- I didn't know about this either, and I support propagating this information to the entire team and enforcing this on PR checks

Reply:
Sounds good. There's a lot of instances of this in existing code which will have to be changed before this can be enabled in PR checks.

Reply:
I'm happy to take another evening next week while watching a show to make these changes like I did for the docstrings. It's satisfying work that doesn't take as much thought.

Reply:
If you decide to, this will filter warnings to those related to lazy formatting for logging.

`./lint --disable=W --enable=logging-not-lazy,logging-format-interpolation,logging-fstring-interpolation`